test/Modules/UngradedModules.jl

```diff
@@ -0,0 +1,675 @@
+  elems = [M3(Oscar.sparse_row(R[0 6 0])),M3(Oscar.sparse_row(R[9 0 -x])),M3(Oscar.sparse_row(R[0 0 -42]))]
+  S3,i3 = sub(M3,elems,:store)
+
+  @test S3 == M3
+  for k=1:5
+      elem = S3(Oscar.sparse_row(array_to_matrix([randpoly(R) for _=1:1, i=1:3])))
+      @test i3(elem) == Oscar.map_canonically(M3, elem)
+  end
+end
+
+@testset "Hom module" begin
+	R, (x0,x1,x2,x3,x4,x5) = PolynomialRing(QQ, ["x0", "x1", "x2", "x3", "x4", "x5"])
+	f1=R[-x2*x3 -x4*x5 0; x0*x1 0 -x4*x5; 0 x0*x1 -x2*x3]'
+	g1=R[x0*x1 x2*x3 x4*x5]'
+	M = cokernel(f1)
+	N = cokernel(g1)
+	SQ = Oscar.hom(M,N)[1]
+
+	#println("-------")
+	#display(SQ)
+	#println("--------")
+	#display(simplify(SQ)[1])
+	#println("--------")
+
+	f2 = R[0 0]
+	M = cokernel(f1)
+	N = cokernel(f2)
+	SQ = hom(M,N)[1]
+
+	#println("-------")
+	#display(SQ)
+	#println("--------")
+	#display(simplify(SQ)[1])
+	#println("--------")
+
+	function free_module_SQ(ring,n)
+		A = one(AbstractAlgebra.MatrixSpace(ring,n,n))
+		B = zero(AbstractAlgebra.MatrixSpace(ring,1,n))
+		return Oscar.SubQuo(A,B)
+	end
+
+	function free_module_SQ(F::Oscar.FreeMod)
+		return Oscar.SubQuo(F,gens(F))
+	end
+
+	# test if Hom(R^n,R^m) gives R^(m*n): (there is a dedicated function for free modules but this is a simple test for the function for subquos)
+	for n=1:5
+		for m=1:5
+			M=free_module_SQ(R,n)
+			N=free_module_SQ(R,m)
+			#SQ = Oscar.hom(M,N)[1] # it's weird that it fails without simplifying
+			SQ = Oscar.simplify(Oscar.hom(M,N)[1])[1]
+			@test SQ == free_module_SQ(Oscar.free_module(SQ))
+		end
+	end
+	R, (x,y) = PolynomialRing(QQ, ["x", "y"])
+	A1 = R[x^2+1 x*y; x^2+y^3 x*y]
+	B1 = R[x+x^4+y^2+1 x^5; y^4-3 x*y^2-1]
+	M1 = Oscar.SubQuo(A1, B1)
+	A2 = R[x;]
+	B2 = R[y^3;]
+	M2 = Oscar.SubQuo(A2,B2)
+	SQ = Oscar.hom(M1,M2)[1]
+	for v in Oscar.gens(SQ)
+		@test v == Oscar.homomorphism_to_module_elem(SQ, Oscar.homomorphism(v))
+	end
+
+	# test if hom(zero-module, ...) is zero
+	Z = Oscar.FreeMod(R,0)
+	@test iszero(Oscar.hom(Z,Z)[1])
+	for k=1:10
+		A = array_to_matrix([randpoly(R,0:15,2,1) for i=1:3,j=1:2])
+		B = array_to_matrix([randpoly(R,0:15,2,1) for i=1:1,j=1:2])
+		N = Oscar.SubQuo(A,B)
+
+		@test iszero(hom(N,Z)[1])
+		@test iszero(hom(Z,N)[1])
+	end
+
+	# test welldefinedness of randomly generated homomorphisms (using hom() and homomorphism())
+	for k=1:10
+		A1 = array_to_matrix([randpoly(R,0:15,2,1) for i=1:3,j=1:2])
+		A2 = array_to_matrix([randpoly(R,0:15,2,1) for i=1:2,j=1:2])
+		B1 = array_to_matrix([randpoly(R,0:15,2,1) for i=1:1,j=1:2])
+		B2 = array_to_matrix([randpoly(R,0:15,2,1) for i=1:1,j=1:2])
+		N = Oscar.SubQuo(A1,B1)
+		M = Oscar.SubQuo(A2,B2)
+		HomNM = Oscar.hom(N,M)[1]
+		for l=1:10
+			H = HomNM(Oscar.sparse_row(array_to_matrix([randpoly(R,0:15,2,1) for _=1:1, j=1:AbstractAlgebra.ngens(HomNM)])))
+			H = Oscar.homomorphism(H)
+			@test Oscar.iswelldefined(H)
+		end
+	end
+end
+
+#=@testset "tensoring morphisms" begin
```

Comment:
Why is this commented out? Perhaps insert a reason, suhc as "this is not yet done/implemented" or "there is a bug in ..."